readthedocs/projects/views/private.py

```diff
@@ -747,8 +747,13 @@ class DomainUpdate(DomainMixin, UpdateView):
     pass
 
 
-class DomainDelete(DomainMixin, DeleteView):
-    pass
+class DomainDelete(DomainMixin, View):
+
+    # POST request is required for the deletion of the domain.
+    def post(self, request, *args, **kwargs):
+        domain = get_object_or_404(Domain, pk=kwargs.get('domain_pk'))
+        domain.delete()
+        return HttpResponseRedirect(self.get_success_url())
```

Comment:
I'm still confused by this view usage, I don't think this should be necessary. I think the problem you're hitting is that you are using an `<a>` element to link to the view view GET. Why not use a form and link to the form via POST?